packages/donations-app/app/components/my-impact/myImpactController.js

```diff
@@ -36,37 +36,77 @@ angular.module('aliceApp')
               vm.charity = vm.project.charity;
             }
 
-            vm.project.allImpactsForProject.forEach((elem) => {
-              let impact = vm.project.impacts.find((e) => {
-                if (e._id === elem._id) {
-                  return e
-                }
-              });
-							if(impact) {
-								elem.userSpent = impact.total;
-								elem.userPercentage = Math.floor(100 * impact.total / elem.target);
-								elem.totalPercentage = Math.floor(100 * elem.totalSpent / elem.target);
-							} else {
-								elem.userSpent = 0;
-								elem.userPercentage = 0;
-								elem.totalPercentage = 0;
-							}
+            // Calculating unitsToHelp and unitHelped
+            vm.project.unitsHelped = vm.project.completedValidations.length;
+            vm.project.totalUnitsToHelp = vm.project.outcomes.reduce((acc, outcome) => {
+              return (Number(outcome.quantityOfUnits) || 0) + acc;
+            }, 0);
+            vm.project.unitsHelpedWithUserDonations = vm.project.impacts.length;
 
-							elem.lightColor = convertHex(elem.color, 0.4);
-							// For stacked progress
-							elem.stacked = [
-								{value: (elem.totalPercentage - elem.userPercentage), color: elem.color},
-								{value: elem.userPercentage, color: "#1998a2"}
-							];
+            // Calculating totalCost per each outcome
+            vm.project.outcomes.forEach(elem => {
+              elem.totalCost = (elem.quantityOfUnits || 0) * (elem.costPerUnit || 0);
+              if (elem.totalCost == elem.moneyUsed) {
```

Comment:
Have you considered calculating the number of outcomes validated directly (I assume we can do it quite easy as we already calculate the total value). 
The added value is the ability to correctly display products without explicit quantityOfUnits